pkg/controller/secret/secret_controller.go

```diff
@@ -89,6 +73,184 @@ type ReconcileSecret struct {
+		Description:    `{{ .CommonLabels.alertname }} {{ .CommonLabels.severity | toUpper }} ({{ len .Alerts }})`,
+		Details: map[string]string{
+			"link":         `{{ if .CommonAnnotations.link }}{{ .CommonAnnotations.link }}{{ else }}https://github.com/openshift/ops-sop/tree/master/v4/alerts/{{ .CommonLabels.alertname }}.md{{ end }}`,
+			"link2":        `{{ if .CommonAnnotations.runbook }}{{ .CommonAnnotations.runbook }}{{ else }}{{ end }}`,
+			"group":        `{{ .CommonLabels.alertname }}`,
+			"component":    `{{ .CommonLabels.alertname }}`,
+			"num_firing":   `{{ .Alerts.Firing | len }}`,
+			"num_resolved": `{{ .Alerts.Resolved | len }}`,
+			"resolved":     `{{ template "pagerduty.default.instances" .Alerts.Resolved }}`,
+		},
+	}
+
+}
+
+func getPagerdutyReceivers(pagerdutyRoutingKey string) []*alertmanager.Receiver {
+	if pagerdutyRoutingKey == "" {
+		return []*alertmanager.Receiver{}
+	}
+
+	receivers := []*alertmanager.Receiver{
+		{
+			Name: receiverNull,
+		},
+		{
+			Name:             receiverPagerduty,
+			PagerdutyConfigs: []*alertmanager.PagerdutyConfig{getPagerdutyConfig(pagerdutyRoutingKey)},
+		},
+	}
+
+	// make-it-warning overrides the severity
+	pdconfig := getPagerdutyConfig(pagerdutyRoutingKey)
+	pdconfig.Severity = "warning"
+	receivers = append(receivers, &alertmanager.Receiver{
+		Name:             receiverMakeItWarning,
+		PagerdutyConfigs: []*alertmanager.PagerdutyConfig{pdconfig},
+	})
+
+	return receivers
+}
+
+func getWatchdogRoute() *alertmanager.Route {
+	return &alertmanager.Route{
+		Receiver:       receiverWatchdog,
+		RepeatInterval: "5m",
+		Match:          map[string]string{"alertname": "Watchdog"},
+	}
+}
+
+func getWatchdogReceivers(watchdogURL string) []*alertmanager.Receiver {
+	if watchdogURL == "" {
+		return []*alertmanager.Receiver{}
+	}
+
+	snitchconfig := &alertmanager.WebhookConfig{
+		NotifierConfig: alertmanager.NotifierConfig{VSendResolved: true},
+		URL:            watchdogURL,
+	}
+
+	return []*alertmanager.Receiver{
+		{
+			Name:           receiverWatchdog,
+			WebhookConfigs: []*alertmanager.WebhookConfig{snitchconfig},
+		},
+	}
+}
+
+func getAlertManagerConfig(pagerdutyRoutingKey string, watchdogURL string) *alertmanager.Config {
+	routes := []*alertmanager.Route{}
+	receivers := []*alertmanager.Receiver{}
+
+	if pagerdutyRoutingKey != "" {
+		routes = append(routes, getPagerdutyRoute())
+		receivers = append(receivers, getPagerdutyReceivers(pagerdutyRoutingKey)...)
+	}
+
+	if watchdogURL != "" {
+		routes = append(routes, getWatchdogRoute())
+		receivers = append(receivers, getWatchdogReceivers(watchdogURL)...)
+	}
+
+	amconfig := &alertmanager.Config{
+		Global: &alertmanager.GlobalConfig{
+			ResolveTimeout: "5m",
+			PagerdutyURL:   pagerdutyURL,
+		},
+		Route: &alertmanager.Route{
+			Receiver: defaultReceiver,
+			GroupByStr: []string{
+				"job",
+			},
+			GroupWait:      "30s",
```

Comment:
Shouldn't we keep those parameters as global constants?

Reply:
We could but didn't because it isn't something I expect we'll change.  The old controller code didn't manage the global config at all.  It merged the bits we wanted into the default provided by OCP.  This PR change things to take 100% control of the alertmanager-main secret.